src/integration/resources/inprocess/coordinator.go

```diff
@@ -166,6 +166,34 @@ func NewCoordinator(cfg config.Configuration, opts CoordinatorOptions) (resource
 	return coord, nil
 }
 
+// NewEmbeddedCoordinator creates a coordinator from one embedded within an existing
+// db node. This method expects that the DB node has already been started before
```

Comment:
How is this expectation guaranteed? Does it error? Should it error if it doesn't?

Reply:
Lazily at the moment. In other words, coordinator API calls will fail since the http server isn't up to handle them. We can validate that here, but that'll take some rework. Will address in a follow up